ycsb-rs/src/main.rs

```diff
@@ -0,0 +1,159 @@
+
+    let config = Arc::new(OBKVClientInitStruct::new(&props));
+
+    if opt.commands.is_empty() {
+        // print commands
+        bail!("no command specified");
+    }
+
+    let database = opt.database.clone();
+    let thread_operation_count = props.operation_count as usize / opt.threads;
+    let actual_client_count = opt.threads / props.obkv_client_reuse;
+    for cmd in opt.commands {
+        let start = Instant::now();
+        let mut threads = vec![];
+        println!(
+            "Database: {database}, Command: {cmd}, Counts Per Threads: {thread_operation_count}"
+        );
+        println!(
+            "Actual Client Count: {actual_client_count}, Client Reuse Count: {}",
+            props.obkv_client_reuse
+        );
+        if database.eq_ignore_ascii_case("obkv") {
+            for _client_idx in 0..actual_client_count {
+                let database = database.clone();
+                let db = db::create_ob(&database, config.clone()).unwrap();
+                for _ in 0..props.obkv_client_reuse {
+                    let db = db.clone();
+                    let wl = wl.clone();
+                    let cmd = cmd.clone();
+                    threads.push(thread::spawn(move || {
+                        let rng = Rc::new(RefCell::new(SmallRng::from_entropy()));
+                        db.init().unwrap();
+                        match &cmd[..] {
+                            "load" => load_ob(wl.clone(), db, thread_operation_count),
+                            "run" => run_ob(wl.clone(), db, rng, thread_operation_count),
+                            cmd => panic!("invalid command: {cmd}"),
+                        };
+                    }));
+                }
+            }
+        } else {
+            for _ in 0..opt.threads {
+                let database = database.clone();
+                let wl = wl.clone();
+                let config = config.clone();
+                let cmd = cmd.clone();
+                threads.push(thread::spawn(move || {
+                    let db = db::create_db(&database, config.clone()).unwrap();
+                    let rng = Rc::new(RefCell::new(SmallRng::from_entropy()));
+
+                    db.init().unwrap();
+
+                    match &cmd[..] {
+                        "load" => load(wl.clone(), db, thread_operation_count),
+                        "run" => run(wl.clone(), db, rng, thread_operation_count),
+                        cmd => panic!("invalid command: {cmd}"),
+                    };
+                }));
+            }
+        }
+        for t in threads {
+            let _ = t.join();
+        }
+        let runtime = start.elapsed().as_millis();
+        println!("[OVERALL], ThreadCount, {}", opt.threads);
+        println!("[OVERALL], RunTime(ms), {runtime}");
+        let throughput = props.operation_count as f64 / (runtime as f64 / 1000.0);
+        println!("[OVERALL], Throughput(ops/sec), {throughput}\n");
```

Comment:
Is this newline still needed even if the print**ln** is used here:
```suggestion
        println!("[OVERALL], Throughput(ops/sec), {throughput}");
```